dist/tools/headerguards/check.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# Copyright 2017 Oliver Hahm <oliver.hahm@inria.fr>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+# customizable
+DIRS="core drivers pkg sys"
+
+# prepare
+EXIT_CODE=0
+
+# check files
+if FILES=$(git grep -lE 'ifndef\ +(_.*_H$|.*_H_$)' ${DIRS})
```

Comment:
> What's the advantage?

Avoiding weird false-positives like e.g.

```C
const struct someweirdtypenameifndef _MY_PRIVATE_CONSTANT_FOR_HEIGHT_H
```

> My assessment was exactly the opposite. Too much cluttered information for my taste.

I'm already annoyed about the doc-checker, that tells me there is a problem in file x, but not what the problem is ;-)